engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -842,6 +878,707 @@ static int SetTexture(lua_State* L)
+
+            lua_pop(L, 1);
+
+            // Check that each stream has the correct amount of components
+            #define VALIDATE_STREAM_LENGTH(stream_name, stream_length, num_components) \
+                if (stream_length % num_components != 0) \
+                { \
+                    lua_pop(L, 2); \
+                    DestroyTextureSet(texture_set_ddf); \
+                    return DM_LUA_ERROR("Uneven number of entries in %s table for geometry [%d]", stream_name, i); \
+                }
+            VALIDATE_STREAM_LENGTH("vertices", geometry.m_Vertices.m_Count, 2);
+            VALIDATE_STREAM_LENGTH("uvs", geometry.m_Uvs.m_Count, 2);
+            VALIDATE_STREAM_LENGTH("indices", geometry.m_Indices.m_Count, 3);
+            #undef VALIDATE_STREAM_LENGTH
+
+            // Calculate extents so that we can transform to -0.5 .. 0.5 based
+            // on the middle of the sprite
+            float geo_width = 0.0f;
+            float geo_height = 0.0f;
+            for (int j = 0; j < geometry.m_Vertices.m_Count; j += 2)
+            {
+                geo_width  = dmMath::Max(geo_width, geometry.m_Vertices.m_Data[j]);
+                geo_height = dmMath::Max(geo_height, geometry.m_Vertices.m_Data[j+1]);
+            }
+
+            geometry.m_Width  = geo_width;
+            geometry.m_Height = geo_height;
+
+            // Transform from texel to local space for position and uvs
+            // Position and texcoords are flipped on y/t axis so that coordinates are
+            // 0,0 in the top left corner, which is the same as the pipeline
+            for (int j = 0; j < geometry.m_Vertices.m_Count; j += 2)
+            {
+                geometry.m_Vertices[j]     = geometry.m_Vertices[j] / geo_width - 0.5;
+                geometry.m_Vertices[j + 1] = 1.0 - (geometry.m_Vertices[j + 1] / geo_height) - 0.5;
+            }
+
+            for (int j = 0; j < geometry.m_Uvs.m_Count; j += 2)
+            {
+                geometry.m_Uvs[j]     = geometry.m_Uvs[j] * inv_tex_width;
+                geometry.m_Uvs[j + 1] = 1.0 - geometry.m_Uvs[j + 1] * inv_tex_height;
+            }
+
+            frame_index_count++;
+        }
+        lua_pop(L, 1); // geometries
+    }
+
+    if (num_animations > 0)
+    {
+        lua_getfield(L, -1, "animations");
+        for (int i = 0; i < num_animations; ++i)
+        {
+            lua_pushnumber(L, i+1);
+            lua_gettable(L, -2);
+            assert(lua_istable(L, -1));
+
+            dmGameSystemDDF::TextureSetAnimation& animation = texture_set_ddf.m_Animations[i];
+
+            // Default values taken from texture_set_ddf.proto
+            animation.m_Fps      = 30;
+            animation.m_Playback = dmGameSystemDDF::PLAYBACK_ONCE_FORWARD;
+
+            lua_getfield(L, -1, "id");
+            animation.m_Id = lua_tostring(L, -1);
+            lua_pop(L, 1);
```

Comment:
I'm wondering if we want to change this to a hash at this point or later.
I guess we can do it later.
From user stand point, they're likely to always use strings, but in the engine, I think it makes little sense to carry around a string for the id.